### PR TITLE
Improvement/workflow gallery improvements

### PR DESCRIPTION
**Pull request by michaelfuest**
## 📦 Pull Request Summary

This PR introduces a series of small frontend features in the workflow step gallery / playground, as well as improvements to UX and small frontend bug fixes.

### ✨ What’s Changed

**Workflow Playground Bug Fixes and Improvements:**
- Made improvements to workflow step config change tracking and status management; went from onEdit behavior to hash-based comparison to prevent unwanted status pending resets due to re-rendering, self-healing or other edge cases
- We now immediately populate step outputs when running test workflow as opposed to only at the end once all steps have been completed, so users see step outputs in real time
- Fixed scrollbar behavior that was messing with the number of cards displayed in the workflow step gallery
- We now have a download button for step inputs, step outputs and final result with a size limit of 100 MB. This is huge for debugging! You can now download step inputs as a json file and debug final transforms / other workflow steps in cursor by executing code on the json that you can now easily extract from sg
- There were a couple of other subtle bugs around input and response schemas being disabled by default despite being saved in the workflow which are now also fixed

---

## 🎯 Motivation / Context

Showing step outputs in real time and making them downloadable is another important step for improving debugging capabilities and dev experience, the rest were subtle frontend bugs and issues that needed fixing.

---

## ✅ Contributor Checklist

- [x] Documentation is up-to-date
- [x] MCP tool descriptions and instructions are up-to-date
- [x] Client SDK updated (if applicable)
- [] Integration test suite has been run and passes (may not be necessary)
- [ ] Unit tests added or updated (if applicable)
- [x] Changes are covered by tests

---

## ⚠️ Compatibility Notes

- [x] ✅ Fully backward compatible
- [ ] ⚠️ Minor behavioral change (non-breaking)
- [ ] ❌ Breaking change — migration steps documented below

<details>
<summary>🔁 Migration Notes (if applicable)</summary>

</details>

---

### CODE REVIEW

File: packages/web/src/app/globals.css [hunk 1]
```diff
@@ -5,6 +5,13 @@
 body {
   overflow: hidden;
   font-family: var(--font-jetbrains-sans);
+  scrollbar-gutter: stable both-edges;
+}
+
+/* Stable scrollbars for scrollable sections to prevent horizontal layout shift */
+.stable-scrollbar,
+[data-stable-scrollbar="true"] {
+  scrollbar-gutter: stable both-edges;
```
Comment (michaelfuest) [inline packages/web/src/app/globals.css:14]: This is to avoid scrollbars slightly changing the width of the step spotlight, which can make the width of the gallery more narrow so that you jump from displaying 3 to 4 cards which makes tab changes from step config to step output feel janky, see https://linear.app/superglue/issue/GLUE-47/investigate-potential-workflow-builder-page-ui-bugs second issue

File: packages/web/src/lib/download-utils.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+
+export const DEFAULT_MAX_DOWNLOAD_BYTES = 100 * 1024 * 1024; // 100 MB
```
Comment (michaelfuest) [inline packages/web/src/lib/download-utils.ts:2]: Not sure about this limit, there may well be intermediate step results that are much larger than this. Could be that this stringification of gigantic JSONs crashes the frontend, so leaving this in for now.

**Comment by heushreck** [comment]
I like the changes, state management works perfect now. 

I have 3 comments:

1. If a step failed, I see the error in the step output - when I look at a different step and go back to the failed step, the error message is no longer visible in the step output
2. we could make the step output windows display a custom text while the step is running - something like "currently running, step outputs will be shown shortly"
3. I like the scroll bar inside of the step config (even though it cuts off the right border). But for me - there is still a scrollbar outside which looks kind of funky (it cuts of the save workflow button and the right arrow to navigate the steps) see screenshot.

Screenshot 2025-09-24 at 16 25 59

File: packages/web/src/components/workflow/WorkflowStepGallery.tsx
Comment (heushreck) [inline packages/web/src/components/workflow/WorkflowStepGallery.tsx:1]: Just a super tiny nit:

When changing the chrome window size (width) from small to large, I go from 2 cards shown, to 3, to 4, then to 5 etc.

2,3, and 4 cards look great. Starting from 5 cards it looks a bit cramped, the edge of the cards are cut of by the arrow buttons. Maybe the cardsToShow calculation needs to be adjusted there
